src/main/java/se/liu/ida/hefquin/engine/datastructures/impl/SolutionMappingsHashTableBasedOnOneVar.java

```diff
@@ -0,0 +1,98 @@
+package se.liu.ida.hefquin.engine.datastructures.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingUtils;
+
+import java.util.*;
+
+public class SolutionMappingsHashTableBasedOnOneVar extends SolutionMappingsHashTable{
+    protected final Map<Node, List<SolutionMapping>> map = new HashMap<>();
+    protected final Var joinVar;
+
+    public SolutionMappingsHashTableBasedOnOneVar(final Set<Var> joinVariables) {
+        if (joinVariables.size() != 1){
+            throw new IllegalArgumentException();
+        }else {
+            this.joinVar = joinVariables.iterator().next();
+        }
+    }
+
+    public SolutionMappingsHashTableBasedOnOneVar(final Var... vars) {
+        if (vars.length != 1) {
+            throw new IllegalArgumentException();
+        }
```

Comment:
Why do you actually create these two constructors. I think it makes more sense to create only one such that this one takes only one `Var` as parameter; i.e., something like this:

```
public SolutionMappingsHashTableBasedOnOneVar(final Var joinVar) {
...
```

That way, there is no need to do these checks here and the code that calls this constructor cannot accidentally have a bug of passing the wrong number of join variables to this hash table.